crm_rma_location_rma/models/stock_warehouse.py

```diff
@@ -0,0 +1,180 @@
+
+from openerp import _, api, fields, models
+
+
+class StockWarehouse(models.Model):
+
+    _inherit = "stock.warehouse"
+
+    lot_rma_id = fields.Many2one('stock.location', 'RMA Location')
+    rma_out_type_id = fields.Many2one('stock.picking.type', 'RMA Out Type')
+    rma_in_type_id = fields.Many2one('stock.picking.type', 'RMA In Type')
+    rma_int_type_id = fields.Many2one('stock.picking.type',
+                                      'RMA Internal Type')
+
+    @api.model
+    def create_sequences_picking_types(self, warehouse):
+        """
+        Takes care of create picking types for internal,
+        incoming and outgoing RMA
+        """
+        seq_obj = self.env['ir.sequence']
+        picking_type_obj = self.env['stock.picking.type']
+        # create new sequences
+
+        if not warehouse.rma_in_type_id:
+            in_seq_id = seq_obj.sudo().create(
+                values={
+                    'name': warehouse.name + _(' Sequence in'),
+                    'prefix': warehouse.code + '/RMA/IN/',
+                    'padding': 5
+                })
+        if not warehouse.rma_out_type_id:
+            out_seq_id = seq_obj.sudo().create(
+                values={
+                    'name': warehouse.name + _(' Sequence out'),
+                    'prefix': warehouse.code + '/RMA/OUT/',
+                    'padding': 5
+                })
+        if not warehouse.rma_int_type_id:
+            int_seq_id = seq_obj.sudo().create(
+                values={
+                    'name': warehouse.name + _(' Sequence internal'),
+                    'prefix': warehouse.code + '/RMA/INT/',
+                    'padding': 5
+                })
+
+        wh_stock_loc = warehouse.lot_rma_id
+
+        # fetch customer and supplier locations, for references
+        customer_loc, supplier_loc = self._get_partner_locations()
+
+        # choose the next available color for
+        # the picking types of this warehouse
+        available_colors = [c % 9 for c in range(3, 12)]
+        all_used_colors = self.env['stock.picking.type'].\
+            search_read([('warehouse_id', '!=', False),
+                         ('color', '!=', False)],
+                        ['color'], order='color')
+        # don't use sets to preserve the list order
+        for col in all_used_colors:
+            if col['color'] in available_colors:
+                available_colors.remove(col['color'])
+
+        color = available_colors[0] if available_colors else 0
+
+        # order the picking types with a sequence
+        # allowing to have the following suit for
+        # each warehouse: reception, internal, pick, pack, ship.
+        max_sequence = self.env['stock.picking.type'].\
+            search_read([], ['sequence'], order='sequence desc')
+        max_sequence = max_sequence and max_sequence[0]['sequence'] or 0
+
+        in_type_id = warehouse.rma_in_type_id
+        if not in_type_id:
```

Comment:
Refactor this into loop will be not simpler and possibly decrease readability, it is similar code, but no the same